proposed/2022/dotnet-nuget-config-spec.md

```diff
@@ -0,0 +1,366 @@
+
+When the verbosity level is detailed or diagnostic, the source(NuGet configuration file path) will be show besides the configuration settings.
+
+#### Examples
+
+- Get all the NuGet configuration settings that will be applied, when invoking NuGet command in the current directory.
+
+```
+dotnet nuget config get all
+
+packageSources:
+  clear
+  add key="source1" value="https://test/source1/v3/index.json"
+  add key="source2" value="https://test/source2/v3/index.json"
+
+packageSourceMapping:
+  clear
+  packageSource key="source1"  pattern="microsoft.*","nuget.*"
+  packageSource key="source2"  pattern="system.*"
+
+packageRestore:
+  add key="enabled" value="False"
+  add key="automatic" value="False"
+
+```
+- Get all the NuGet configuration settings that will be applied, when invoking NuGet command in the specific directory.
+
+```
+dotnet nuget config get all C:\Test\Repos
+
+packageSources:
+  clear
+  add key="source1" value="https://test/source1/v3/index.json"
+  add key="source2" value="https://test/source2/v3/index.json"
+
+packageSourceMapping:
+  clear
+  packageSource key="source1"  pattern="microsoft.*","nuget.*"
+  packageSource key="source2"  pattern="system.*"
+
+packageRestore:
+  add key="enabled" value="False"
+  add key="automatic" value="False"
+
+```
+
+- Get all the NuGet configuration settings that will be applied, when invoking NuGet command in the current directory. Show the source(nuget configuration file path) of each configuration settings/child items.
+
+```
+dotnet nuget config get all -v d
+
+packageSources:
+  add key="source1" value="https://test/source1/v3/index.json"     file: C:\Test\Repos\Solution\NuGet.Config
+  add key="source2" value="https://test/source2/v3/index.json"     file: C:\Test\Repos\NuGet.Config
+
+packageSourceMapping:
+  clear                                                            file: C:\Users\username\AppData\Roaming\NuGet\NuGet.Config
+  packageSource key="source1"  pattern="microsoft.*","nuget.*"     file: C:\Test\Repos\Solution\NuGet.Config
+  packageSource key="source2"  pattern="system.*"                  file: C:\Test\Repos\NuGet.Config                                 
+
+packageRestore:
+  add key="enabled" value="False"                                 file: C:\Users\username\AppData\Roaming\NuGet\NuGet.Config
+  add key="automatic" value="False"                               file: C:\Users\username\AppData\Roaming\NuGet\NuGet.Config
+
+```
+
+- Get `http_proxy` from config section, when invoking NuGet command in the current directory.
+
+```
+dotnet nuget config get http_proxy 
+
+http://company-squid:3128@contoso.com"
+
+```
+
+- Get `http_proxy` from config section, when invoking NuGet command in the current directory. Show the source(nuget configuration file path) of this configuration setting.
+
+```
+dotnet nuget config get http_proxy 
```

Comment:
Is there supposed to be a "show path" or verbosity parameter here?

Reply:
Good catch! Added `--show-path`. Thanks!